src/main/java/org/jenkinsci/plugins/envinject/EnvInjectEnvVarsContributor.java

```diff
@@ -0,0 +1,27 @@
+package org.jenkinsci.plugins.envinject;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.model.*;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+@Extension
+public class EnvInjectEnvVarsContributor extends EnvironmentContributor {
+
+    @Override
+    public void buildEnvironmentFor(Job job, EnvVars env, TaskListener listener) throws IOException, InterruptedException {
+        for (Object property : job.getProperties().values()) {
```

Comment:
job.getProperty() may return null if you run the method agains the loaded job, which has been saved before the EnvInject installation. 

<code>job.getProperties()</code> is never null, but IMO you're using a wrong pattern to lookup for properties. It may become performance-innefective in future Jenkins versions